support/util/util.go

```diff
@@ -458,3 +458,109 @@ func PredicatesForHostedClusterChildResourcesAnnotationScoping(r client.Reader)
 	}
 	return predicate.NewPredicateFuncs(filter)
 }
+
+// PredicatesForNodepoolAnnotationScoping returns predicate filters for all event types that will ignore incoming
+// event requests for nodepool resources in which their owning hostedcluster resource doesn't have a scope annotation
+// that matches what is specified in the HOSTEDCLUSTERS_SCOPE_ANNOTATION env var.  If not defined or empty, the default behavior
+// is to accept all nodepool events in which the owning hostedcluster resource does not have a corresponding scope annotation defined.
+// The ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING env var must also be set to "true" to enable the scoping feature.
+func PredicatesForNodepoolAnnotationScoping(r client.Reader) predicate.Predicate {
```

Comment:
If we are using the state of the `HostedCluster` to determine whether or not to process a `NodePool`, we need to re-queue affected `NodePool`s when the `HostedCluster` changes (e.g. it gains or loses the annotation we're checking for)

Reply:
I believe this is already happening, right?  In the hypothetical 2 operator topology, where "operator1" manages hostedclusters that do not have the specified annotation, and "operator2" is configured to manage hosteclusters that do have the annotation, as scope annotations are added/removed from hosteclusters, one of the operators should in-turn queue the NodePools.